rest/data/host_create.go

```diff
@@ -157,6 +162,44 @@ func CreateHostsFromTask(ctx context.Context, env evergreen.Environment, t *task
 	return catcher.Resolve()
 }
 
+// GetDistro returns the distro based on the name and provider.
+func GetDistro(ctx context.Context, createHost apimodels.CreateHost) (*distro.Distro, error) {
```

Comment:
nit: since this is only usable with `host.create`, can we call this something more specific like `GetHostCreateDistro`?